File: leetcode/0071-simplify-path/solution.cpp
string simplifyPath(string path) {
    vector<string> path_stack;
    stringstream ss(path);
    string c;

    while (getline(ss, c, '/')) {
        if (c == "..") {
            if (!path_stack.empty()) {
                path_stack.pop_back();
            }
        } else if (c.size() != 0 && c != ".") {
            path_stack.push_back(c);
        }
    }

    string ans = "";
    for (string dir : path_stack) {
        ans += "/" + dir;
    }

    if (ans.size() == 0) {
        return "/";
    }

    return ans;
}

File: leetcode/0071-simplify-path/solution.py
def simplifyPath(self, path: str) -> str:
    path = path.replace('//', '/')
    path_dirs = path.split('/')
    path_stack = []

    for i, _dir in enumerate(path_dirs):
        if _dir == '..':
            if len(path_stack):
                path_stack.pop()
        elif len(_dir) and _dir != '.':
            path_stack.append(_dir)

    return '/' + '/'.join(path_stack)
